lib/bloc/podcast/podcast_bloc.dart

```diff
@@ -320,6 +321,13 @@ class PodcastBloc extends Bloc {
   void _listenPodcastStateEvents() async {
     _podcastEvent.listen((event) async {
       switch (event) {
+        case PodcastEvent.episodeFilterDownloaded:
```

Comment:
You've added the downloaded filter here which is great. If you look at what this code is doing, it is setting the filter, saving the episode and then re-fetching it. It does this so that the sort and filter is preserved, i.e. saved to disk. Take a look at the `_loadFilteredEpisodes()` method and follow the path through. You'll find that the filter is applied at the bottom database level.